rskj-core/src/main/java/org/ethereum/core/Transaction.java

```diff
@@ -107,27 +118,107 @@ protected Transaction(byte[] rawData) {
             throw new IllegalArgumentException("A transaction must have exactly 9 elements");
         }
 
-        this.nonce = transaction.get(0).getRLPData();
-        this.gasPrice = RLP.parseCoinNonNullZero(transaction.get(1).getRLPData());
-        this.gasLimit = transaction.get(2).getRLPData();
-        this.receiveAddress = RLP.parseRskAddress(transaction.get(3).getRLPData());
-        this.value = RLP.parseCoinNullZero(transaction.get(4).getRLPData());
-        this.data = transaction.get(5).getRLPData();
-
-        // only parse signature in case tx is signed
-        byte[] vData = transaction.get(6).getRLPData();
-        if (vData != null) {
-            if (vData.length != 1) {
-                throw new TransactionException("Signature V is invalid");
+        if (transaction.get(0) instanceof RLPList) {
+            // Transaction format 1
+            this.version = RLP.decodeInt(nullToZeroArray(((RLPList) transaction.get(0)).get(0).getRLPData()), 0);
+            if (this.version != 1) {
+                throw new TransactionException("Transaction has unknown version");
+            }
+            if (transaction.get(1) instanceof RLPList) {
+                List<byte[]> nonce = new ArrayList<>();
+                for (RLPElement element : (RLPList) transaction.get(1)) {
+                    nonce.add(nullToZeroArray(element.getRLPData()));
+                }
+                this.nonce = Collections.unmodifiableList(nonce);
+            } else {
+                this.nonce = Collections.singletonList(nullToZeroArray(transaction.get(1).getRLPData()));
+            }
+            this.gasPrice = RLP.parseCoinNonNullZero(transaction.get(2).getRLPData());
+            this.gasLimit = transaction.get(3).getRLPData();
+            this.receiveAddress = RLP.parseRskAddress(transaction.get(4).getRLPData());
+            this.value = RLP.parseCoinNullZero(transaction.get(5).getRLPData());
+            this.data = transaction.get(6).getRLPData();
+            this.ownerCount = RLP.decodeInt(nullToZeroArray(transaction.get(7).getRLPData()), 0);
+
+            if (transaction.get(8) instanceof RLPList) {
+                RLPList signaturesField = (RLPList) transaction.get(8);
+                if (signaturesField.isEmpty() || signaturesField.get(0) instanceof RLPList) {
+                    List<ECDSASignature> signatures = new ArrayList<>();
+                    byte chainId = 0;
+                    boolean chainIdUnset = true;
+                    for (RLPElement e : signaturesField) {
+                        if (e instanceof RLPList) {
+                            RLPList element = (RLPList) e;
+                            if (element.size() != 3) {
+                                throw new IllegalArgumentException("Signature field must be made up of 3 elements");
+                            }
+                            byte[] vData = element.get(0).getRLPData();
```

Comment:
similar code duplicated below, better to encapsulate signature parsing in a separate method.